addons/survey/models/survey_question.py

```diff
@@ -810,6 +810,8 @@ class SurveyQuestionAnswer(models.Model):
     # question and question related fields
     question_id = fields.Many2one('survey.question', string='Question', ondelete='cascade', index='btree_not_null')
     matrix_question_id = fields.Many2one('survey.question', string='Question (as matrix row)', ondelete='cascade', index='btree_not_null')
+    question_survey_id = fields.Many2one(related='question_id.survey_id', string="Question Survey id", store=True)
+    matrix_question_survey_id = fields.Many2one(related='matrix_question_id.survey_id', string="Matrix Question Survey id", store=True)
```

Comment:
why do we need related stored ?

> Also, we add a 'survey_id' field to the 'survey_question_answer' model
so as to simplify the domain in the related record rule.

if it's just to simplify the domain it's not worth imo (or at least do not need to be stored)

Reply:
Ok, I 'unstore' the fields :+1: